File: app/src/main/java/com/soft/bitna/kopper/ForRecyclerViews/HochaRecyclerViewAdapter.java
package com.soft.bitna.kopper.ForRecyclerViews;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.soft.bitna.kopper.R;

import java.util.ArrayList;

public class HochaRecyclerViewAdapter extends RecyclerView.Adapter<HochaRecyclerViewAdapter.HochaViewHolder> {

    Context context;
    ArrayList<String> list;

    public HochaRecyclerViewAdapter(Context context, ArrayList<String> list) {
        super();
        this.context = context;
        this.list = list;
    }

    @Override
    public void onBindViewHolder(HochaRecyclerViewAdapter.HochaViewHolder holder, int position) {
        holder.hochaTv.setText(list.get(position));
    }

    @Override
    public HochaRecyclerViewAdapter.HochaViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new HochaRecyclerViewAdapter.HochaViewHolder(LayoutInflater.from(context).inflate(R.layout.hocha_textview, parent, false));
    }

    @Override
    public int getItemCount() {
        if(list!=null) {
            return list.size();
        }
        else{
            return 0;
        }
    }

    public class HochaViewHolder extends RecyclerView.ViewHolder{
        TextView hochaTv;

        public HochaViewHolder(View itemView) {
            super(itemView);

            hochaTv = itemView.findViewById(R.id.hochaTv);
        }
    }

}

File: app/src/main/java/com/soft/bitna/kopper/ForRecyclerViews/HochaRecyclerViewAdapter.kt
package com.soft.bitna.kopper.ForRecyclerViews

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.soft.bitna.kopper.R

import java.util.ArrayList

class HochaRecyclerViewAdapter(internal var context: Context, internal var list: ArrayList<String>?) : RecyclerView.Adapter<HochaRecyclerViewAdapter.HochaViewHolder>() {

    override fun onBindViewHolder(holder: HochaRecyclerViewAdapter.HochaViewHolder, position: Int) {
        holder.hochaTv.text = list!![position]
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HochaRecyclerViewAdapter.HochaViewHolder {
        return HochaRecyclerViewAdapter.HochaViewHolder(LayoutInflater.from(context).inflate(R.layout.hocha_textview, parent, false))
    }

    override fun getItemCount(): Int {
        return if (list != null) {
            list!!.size
        } else {
            0
        }
    }

    inner class HochaViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var hochaTv: TextView

        init {

            hochaTv = itemView.findViewById(R.id.hochaTv)
        }
    }

}
